### PR TITLE
bpo-30853: IDLE: Convert font and general vars to use VarTrace

### PR DESCRIPTION
**Pull request by csabella**
https://bugs.python.org/issue30853

### CODE REVIEW

File: Lib/idlelib/configdialog.py [hunk 1]
```diff
@@ -76,12 +77,14 @@ def __init__(self, parent, title='', _htest=False, _utest=False):
...
         self.load_configs()
         self.attach_var_callbacks()  # Avoid callbacks during load_configs.
+        tracers.attach()
 
         if not _utest:
             self.grab_set()
             self.wm_deiconify()
             self.wait_window()
 
+
```
Comment (mlouielu) [inline Lib/idlelib/configdialog.py:87]: Class method only need one blank line, so this is no need to changed.

**Review by csabella — COMMENTED** [review]
Thanks!   Global makes more sense.

File: Lib/idlelib/configdialog.py [hunk 2]
```diff
@@ -518,12 +486,17 @@ def create_page_highlight(self):
             'Shell Stderr Text': ('stderr', '13'),
             }
         parent = self.parent
-        self.builtin_theme = StringVar(parent)
-        self.custom_theme = StringVar(parent)
+        self.builtin_theme = tracers.add(
+                StringVar(parent), self.var_changed_builtin_theme)
+        self.custom_theme = tracers.add(
+                StringVar(parent), self.var_changed_custom_theme)
         self.fg_bg_toggle = BooleanVar(parent)
```
Comment (terryjreedy) [inline Lib/idlelib/configdialog.py:493]: It is now obvious that fg_bg_toggle is not traced, others are.  A plus for the change.

**Review by terryjreedy — APPROVED** [review]

File: Lib/idlelib/configdialog.py [hunk 3]
```diff
@@ -75,13 +76,15 @@ def __init__(self, parent, title='', _htest=False, _utest=False):
         # self.bind('<Alt-a>', self.Apply) #apply changes, save
         # self.bind('<F1>', self.Help) #context help
         self.load_configs()
-        self.attach_var_callbacks()  # Avoid callbacks during load_configs.
+        # Avoid callbacks during load_configs.
```
Comment (terryjreedy) [inline Lib/idlelib/configdialog.py:79]: We must remember this when reviewing and writing tests for the load functions.

**Pull request merged**